source/fundamentals/enterprise-auth.txt

```diff
@@ -280,3 +287,154 @@ mechanism:
+support.
+
+You can specify GCP IMDS OIDC authentication either by 
+using a ``MongoCredential`` object or as part of the connection string. Select the 
+:guilabel:`Connection String` or :guilabel:`MongoCredential` tab to
+see the corresponding syntax.
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: mongodb-gcp-imds-connection-string
+    
+      Replace the ``<resource>`` placeholder with the percent-encoded application or
+      service that the OIDC access token is intended for.
+     
+      .. code-block:: java
+
+         MongoClient mongoClient = MongoClients.create("mongodb://<username>@<hostname>:<port>/?authMechanism=MONGODB-OIDC&authMechanismProperties=ENVIRONMENT:gcp,TOKEN_RESOURCE:<resource>");
+
+   .. tab:: MongoCredential
+      :tabid: mongodb-gcp-mongo-credential
+
+      Replace the ``<username>`` placeholder with the client ID of the GCP managed
+      identity. Replace the ``<resource>`` placeholder with the
+      application or service that the OIDC access token is intended for.
+      You must percent-encode the resource value
+      if it contains a comma (``,``), plus sign (``+``), or ampersand (``&``). 
+      
+      .. code-block:: java
+
+         MongoCredential credential = MongoCredential.createOidcCredential("<username>")
+            .withMechanismProperty("TOKEN_RESOURCE", "<resource>")
+            .withMechanismProperty("ENVIRONMENT", "gcp");  
+   
+         MongoClient mongoClient = MongoClients.create(
+             MongoClientSettings.builder()
+                 .applyToClusterSettings(builder ->
+                         builder.hosts(Arrays.asList(new ServerAddress("<hostname>", <port>))))
+                 .credential(credential)
+                 .build());
+
+Custom Callback
++++++++++++++++
+
+On some platforms, such as Azure Functions and Azure Kubernetes Service (AKS), you
+must define a custom callback method to use OIDC to authenticate. To use a custom
+callback to retrieve an OIDC token, call the ``withMechanismProperty()`` method on
+your ``MongoCredential`` object to set the following options:
+
+- ``"ALLOWED_HOSTS"``: The list of allowed hostnames or IP addresses for MongoDB.
+  The hostnames may include ``"*."`` as a leading wildcard, which allows for
+  matching subdomains. By default, this property includes ``"*.mongodb.net"``,
+  ``"*.mongodb-qa.net"``, ``"*.mongodb-dev.net"``, ``"*.mongodbgov.net"``, ``"localhost"``,
+  ``"127.0.0.1"``, and ``"::1"``.
+- ``"OIDC_CALLBACK"``: The method that retrieves the OIDC token. This method must accept
+  a ``context`` object as a parameter and return an ``OidcCallbackResult`` object.
+
+The following code example shows how to include
+a callback method in your ``MongoCredential`` object. In this example, the callback
+retrieves an OIDC token from a file named ``"access-token.dat"`` in the local file system.
+
+.. code-block:: java
+
+   MongoCredential credential = new Credential()
```

Comment:
This needs to be a `MongoCredential.createOidcCredential(null)`

Reply:
Thanks--something felt off about it.